File: android/sample-ui/src/main/kotlin/com/usesmileid/sampleapps/ui/screens/ProfileSwitchSheet.kt
package com.usesmileid.sampleapps.ui.screens

import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import com.usesmileid.sampleapps.ui.UseSmileIDSampleTestIds
import com.usesmileid.sampleapps.ui.components.UseSmileIDSampleBottomSheet
import com.usesmileid.sampleapps.ui.components.UseSmileIDSampleProfileRow
import com.usesmileid.sampleapps.ui.components.avatarColorForProfile
import com.usesmileid.sampleapps.ui.state.UseSmileIDSampleProfile

/** The profile-switch sheet. Selecting one switches immediately, which is why it needs no save action. */
@Composable
fun ProfileSwitchSheet(
    profiles: List<UseSmileIDSampleProfile>,
    activeId: String?,
    onSelect: (UseSmileIDSampleProfile) -> Unit,
    onDismissRequest: () -> Unit,
    modifier: Modifier = Modifier,
    /** Null hides the row, for a host that offers no way to create one. */
    onCreate: (() -> Unit)? = null,
) {
    UseSmileIDSampleBottomSheet(
        onDismissRequest = onDismissRequest,
        modifier = modifier,
        title = "Switch profile",
        testId = UseSmileIDSampleTestIds.PROFILE_SWITCH_SHEET,
    ) {
        profiles.forEachIndexed { index, profile ->
            UseSmileIDSampleProfileRow(
                avatarColor = avatarColorForProfile(index),
                organisation = profile.title,
                supportingText = profile.caption,
                initials = profile.initials,
                selected = profile.id == activeId,
                onClick = { onSelect(profile) },
                testId = UseSmileIDSampleTestIds.profileRow(profile.id),
            )
        }
        if (onCreate != null) {
            UseSmileIDSampleProfileRow(
                organisation = "New profile",
                supportingText = "Run jobs as someone else",
                initials = "",
                selected = false,
                onClick = onCreate,
                testId = UseSmileIDSampleTestIds.PROFILE_SWITCH_NEW,
            )
        }
    }
}

File: ios/SampleUI/Sources/SampleUI/Screens/ProfileSwitchSheet.swift
import SwiftUI

/// The profile-switch sheet. Selecting one switches immediately, which is why it needs no save action.
public struct ProfileSwitchSheet: View {
  private let profiles: [UseSmileIDSampleProfile]
  private let activeId: String?
  private let onSelect: (UseSmileIDSampleProfile) -> Void
  private let onCreate: (() -> Void)?

  /// A nil `onCreate` hides the "New profile" row, for a host that offers no way to create one.
  public init(
    profiles: [UseSmileIDSampleProfile],
    activeId: String?,
    onSelect: @escaping (UseSmileIDSampleProfile) -> Void,
    onCreate: (() -> Void)? = nil
  ) {
    self.profiles = profiles
    self.activeId = activeId
    self.onSelect = onSelect
    self.onCreate = onCreate
  }

  public var body: some View {
    UseSmileIDSampleBottomSheet(title: "Switch profile", testId: UseSmileIDSampleTestIds.profileSwitchSheet) {
      ForEach(Array(profiles.enumerated()), id: \.element.id) { index, profile in
        UseSmileIDSampleProfileRow(
          organisation: profile.title,
          supportingText: profile.caption,
          initials: profile.initials,
          selected: profile.id == activeId,
          avatarColor: useSmileIDSampleAvatarColor(profileIndex: index),
          testId: UseSmileIDSampleTestIds.profileRow(profile.id),
          onTap: { onSelect(profile) }
        )
      }
      if let onCreate {
        UseSmileIDSampleProfileRow(
          organisation: "New profile",
          supportingText: "Run jobs as someone else",
          initials: "",
          selected: false,
          testId: UseSmileIDSampleTestIds.profileSwitchNew,
          onTap: onCreate
        )
      }
    }
  }
}
